cleanlab/multiannotator.py

```diff
@@ -0,0 +1,928 @@
+        Specifies the method used to calculate the quality of the consensus label.
+        For valid quality methods, view :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`
+
+    Returns
+    -------
+    annotator_stats : pd.DataFrame
+        Returns overall statistics about each annotator.
+        For details, see the documentation of :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`.
+    """
+
+    annotator_quality = _get_annotator_quality(
+        labels_multiannotator=labels_multiannotator,
+        pred_probs=pred_probs,
+        consensus_label=consensus_label,
+        num_annotations=num_annotations,
+        annotator_agreement=annotator_agreement,
+        model_weight=model_weight,
+        annotator_weight=annotator_weight,
+        quality_method=quality_method,
+    )
+
+    # Compute the number of labels labeled/ by each annotator
+    num_labeled = labels_multiannotator.count()
+
+    # Compute the fraction of labels annotated by each annotator that agrees with the consensus label
+    # TODO: check if we should drop singleton labels here
+    agreement_with_consensus = labels_multiannotator.apply(
+        lambda s: np.mean(s[pd.notna(s)] == consensus_label[pd.notna(s)]),
+        axis=0,
+    ).to_numpy()
+
+    # Find the worst labeled class for each annotator
+    # TODO: potentially have to do tiebreaking here
+    worst_class = labels_multiannotator.apply(
+        lambda s: (s[pd.notna(s)] == consensus_label[pd.notna(s)]).groupby(s).mean().idxmin()
+    ).astype("int64")
+
+    # Create multi-annotator stats DataFrame from its columns
+    annotator_stats = pd.DataFrame(
+        {
+            "annotator_quality": annotator_quality,
+            "num_labeled": num_labeled,
+            "agreement_with_consensus": agreement_with_consensus,
+            "worst_class": worst_class,
+        }
+    )
+
+    return annotator_stats.sort_values(by=["annotator_quality", "agreement_with_consensus"])
+
+
+def get_label_quality_multiannotator(
+    labels_multiannotator: Union[pd.DataFrame, np.ndarray],
+    pred_probs: np.ndarray,
+    *,
+    consensus_method: Union[str, List[str]] = "best_quality",
+    quality_method: str = "auto",
+    return_detailed_quality: bool = False,
+    return_annotator_stats: bool = False,  # sort by lowest overall_quality first
+    verbose: bool = True,
+    label_quality_score_kwargs: dict = {},
+) -> Union[pd.DataFrame, Dict[str, pd.DataFrame]]:
+    """Returns label quality scores for each example for each annotator.
+    This function is for multiclass classification datasets where examples have been labeled by
+    multiple annotators (not necessarily the same number of annotators per example).
+    It computes quality scores for each annotator's labels and other useful values like how
+    confident we are that the current consensus label is actually correct.
```

Comment:
"like how confident we are that the current consensus label is actually correct"
->
"like one consensus label for each example that best accounts for the labels chosen by each annotator (and their quality), as well as a score for how confident we are that this consensus label is actually correct"